### DISCUSSION TITLE
Nix Support

### DISCUSSION THREAD

**Discussion by NotAShelf** [robherley/snips.sh#71] [Ideas]
Hello! I have recently discovered snips.sh, and after a short while I really enjoyed using it. As such, I would like to package it for my favorite package manager: Nix.

For maintainability reasons, I would like to start (I've already finished a draft package) development and maintenance of a Nix package and a NixOS module in this repository. Would that be possible? Or should I work on PRing snips.sh to nixpkgs?

**Comment by robherley** [comment]
👋 Sorry for the late reply, but it'd be great to get this into a nix package. Unfortunately I don't have _any_ experience with nix.

The only real dependencies are sqlite and libtensorflow, but it looks like those are already on nixpkgs.

**Reply by NotAShelf** [reply]
Thank you for your interest! I've already written a derivation (think of them as PKGBUILDs or RPMs), as well as a flake that allows us to provide a package from the repo, allowing updates in realtime.

Depending on how you wish to proceed, I can open a pull request that introduces in-repo nix support or/and work on getting the package into nixpkgs.

**Comment by robherley** [comment]
In case anyone else stumbles on this thread, snips is available in nixpkgs: https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/sn/snips-sh/package.nix
